ios/browser/api/networking/models/extensions/BraveCertificatePoliciesExtensionModel.mm

```diff
@@ -0,0 +1,186 @@
+  _notice = notice;
+}
+
+- (void)setUnknown:(NSString*)unknown {
+  _unknown = unknown;
+}
+@end
+
+@implementation BraveCertificatePolicyInfoExtensionModel
+- (instancetype)init {
+  if ((self = [super init])) {
+    _oid = [[NSString alloc] init];
+    _qualifiers = @[];
+  }
+  return self;
+}
+
+- (void)setOid:(NSString*)oid {
+  _oid = oid;
+}
+
+- (void)setQualifiers:(NSArray<BraveCertificatePolicyInfoQualifierExtensionModel*>*)qualifiers {
+  _qualifiers = qualifiers;
+}
+@end
+
+@implementation BraveCertificatePoliciesExtensionModel
+- (void)parseExtension:(X509_EXTENSION*)extension {
+  NSMutableArray* result = [[NSMutableArray alloc] init];
+  
+  //STACK_OF(POLICYINFO)
+  CERTIFICATEPOLICIES* policies = static_cast<CERTIFICATEPOLICIES*>(X509V3_EXT_d2i(extension));
+  if (policies) {
+    for (std::size_t i = 0; i < sk_POLICYINFO_num(policies); ++i) {
+      POLICYINFO* info = sk_POLICYINFO_value(policies, static_cast<int>(i));
+      if (info) {
+        auto* policy = [[BraveCertificatePolicyInfoExtensionModel alloc] init];
+        [policy setOid:brave::string_to_ns(x509_utils::string_from_ASN1_OBJECT(info->policyid, true))];
+        
+        if (info->qualifiers) {
+          NSMutableArray* qualifiers = [[NSMutableArray alloc] init];
+          
+          for (std::size_t i = 0; i < sk_POLICYQUALINFO_num(info->qualifiers); ++i) {
+            POLICYQUALINFO* qualifier_info =
+                                  sk_POLICYQUALINFO_value(info->qualifiers, static_cast<int>(i));
+            if (qualifier_info) {
+              auto* info = [[BraveCertificatePolicyInfoQualifierExtensionModel alloc] init];
+              switch (OBJ_obj2nid(qualifier_info->pqualid)) {
+                case NID_id_qt_cps: {
+                  //ASN1_IA5STRING
+                  [info setPqualid:NID_id_qt_cps];
+                  [info setCps:brave::string_to_ns(
+                                        x509_utils::string_from_ASN1STRING(qualifier_info->d.cpsuri))];
+                }
+                  break;
+
+                case NID_id_qt_unotice: {
+                  [info setPqualid:NID_id_qt_unotice];
+                  USERNOTICE* user_notice = qualifier_info->d.usernotice;
+                  if (user_notice) {
+                    auto* notice = [[BraveCertificatePolicyInfoQualifierNoticeExtensionModel alloc] init];
+                    
+                    //i2d_USERNOTICE
+                    if (user_notice->noticeref) {
+                      NOTICEREF* notice_ref = user_notice->noticeref;
+                      [notice setOrganization:brave::string_to_ns(
+                                                  x509_utils::string_from_ASN1STRING(notice_ref->organization))];
+                      
+                      if (notice_ref->noticenos) {
+                        NSMutableArray* noticeNumbers = [[NSMutableArray alloc] init];
+                        for (std::size_t i = 0; i < sk_ASN1_INTEGER_num(notice_ref->noticenos); ++i) {
```

Comment:
redefinition again; given that I know you tested this, maybe the compiler is noticing you're redefining and created a new scoped variable? either way, this could use a fix up 😄